### PR TITLE
feat: add before_closing_*_tag mfa support

### PR DESCRIPTION
**Pull request by yordis**

### CODE REVIEW

File: lib/ex_doc/utils.ex [hunk 1]
```diff
@@ -1,6 +1,14 @@
 defmodule ExDoc.Utils do
   @moduledoc false
 
+  def run_before_closing_body_tag(%{before_closing_body_tag: {m, f, a}}, module) do
+    apply(m, f, a ++ [module])
```
Comment (josevalim) [inline lib/ex_doc/utils.ex:5]:
```suggestion
    apply(m, f, [module | a])
```

Typically we prepend args to MFAs.

Overall this looks good to me!

**Comment by josevalim** [comment]
LGTM! We just need to update the docs and we can ship it!

**Comment by starbelly** [comment]
@josevalim  Maybe this another PR, but wdyt about supporting maps? The case I'm thinking about is when you just have some static data you want appended, making a callback less valuable. 

`%{html: "stuff", epub: "other stuff"}`

**Comment by josevalim** [comment]
Sounds good to me too.

**Comment by yordis** [comment]
ahhh that is what you meant @starbelly 🤦🏻 I can make another PR. Let me find some time to update some docs.

**Comment by yordis** [comment]
Ready again

**Comment by josevalim** [comment]
For some reason the build is not working. Can you please rebase just in case?

**Comment by yordis** [comment]
I rebased, but I am unsure what is happening with CI ...

**Comment by josevalim** [comment]
I will merge and play with it. Thanks!

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**